### PR TITLE
refactor: various cleanups

### PR DESCRIPTION
**Pull request by huitseeker**
Opinion on those refactors loosely held, happy to accomodate. But some of the same was recently merged in #1810, so I thought this rounds things up a bit.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
tysm!

lgtm

File: crates/rpc/rpc-engine-api/src/engine_api.rs [hunk 1]
```diff
@@ -352,32 +348,26 @@ impl<Client: HeaderProvider + BlockProvider + StateProviderFactory + EvmEnvProvi
...
 
-        let head = if let Some(head) = self.client.header(&head_block_hash)? {
-            head
-        } else {
-            // Block is not known, nothing to do.
-            return Ok(ForkchoiceUpdated::from_status(PayloadStatusEnum::Syncing))
-        };
+        let Some(head) = self.client.header(&head_block_hash)? else {
+ // Block is not known, nothing to do
+ return Ok(ForkchoiceUpdated::from_status(PayloadStatusEnum::Syncing)) };
```
Comment (onbjerg) [inline crates/rpc/rpc-engine-api/src/engine_api.rs:353]: Can you indent this? Seems weird that no linter caught it

File: crates/rpc/rpc-engine-api/src/engine_api.rs [hunk 2]
```diff
@@ -352,32 +348,26 @@ impl<Client: HeaderProvider + BlockProvider + StateProviderFactory + EvmEnvProvi
...
         }
 
-        let head_td = if let Some(head_td) = self.client.header_td(&head_block_hash)? {
-            head_td
-        } else {
-            // internal error - we have the head block but not the total difficulty
-            return Ok(ForkchoiceUpdated::from_status(PayloadStatusEnum::Invalid {
+        let Some(head_td) = self.client.header_td(&head_block_hash)? else {
+ // internal error - we have the head block but not the total difficulty
+ return Ok(ForkchoiceUpdated::from_status(PayloadStatusEnum::Invalid {
```
Comment (onbjerg) [inline crates/rpc/rpc-engine-api/src/engine_api.rs:362]: Same here

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Comment by mattsse** [comment]
Oops, sorry for merging this prematurely.

looks like ruftmt has problems formatting let else?

**Comment by onbjerg** [comment]
Looks like it. A bit annoying

**Pull request merged**